src/resolvers.js

```diff
@@ -315,6 +315,9 @@ export class NewInstance {
 }
 
 export function getDecoratorDependencies(target, name) {
+  if (!target.hasOwnProperty('inject')) {
```

Comment:
Is the idea here to effectively block the inheritance chain?

Reply:
yes, as the parameter decorators act as autoinject themself (adding @autoinject would just be for clarity) it needs the same autoinject. and with autoinject, one never wants to use the inherited one. after all, autoinject, takes it all from the current constructor parameters.
i'll rewrite it, to make it clearer, that it is exactly the same